create_openapi_definition.py

```diff
@@ -0,0 +1,213 @@
+        },
+        {
+            "name": "Customers",
+            "description": "Requests against customer resources"
+        },
+        {
+            "name": "Comments",
+            "description": "Requests against comment resources"
+        },
+        {
+            "name": "Attachments",
+            "description": "Requests against attachment resources"
+        },
+        {
+            "name": "CustomerUserAssociations",
+            "description": "Requests against customer-user association resources"
+        }
+    ],
+    # It's nice to have the paths for our definitions in order, starting from Root, since they
+    # will show up in that order on Swaggerhub.
+    "paths": {path_value: dict() for key, path_value in RESOURCE_PATH_LOOKUP.items()}
+}
+
+# Create a YAML object from ruamel.yaml for loading and dumping yaml data.
+# Here we're using the ruamel.yaml library because it handles string block scalars better than
+# pyyaml.
+# A string block scalar in yaml is this:
+#
+# value: |
+#   {
+#       "_links": {
+#           "self": {
+#               "href": "/"
+#           },
+#           "users": {
+#               "href": "/users"
+#           },
+#           "customers": {
+#               "href": "/customers"
+#           },
+#           "tickets": {
+#               "href": "/tickets"
+#           },
+#           "comments": {
+#               "href": "/comments"
+#           },
+#           "attachments": {
+#               "href": "/attachments"
+#           },
+#           "customer_user_associations": {
+#               "href": "/customer_user_associations"
+#           }
+#       }
+#   }
+yaml = YAML()
+
+if __name__ == "__main__":
+    # Recursively go through the openapi_resource_definitions directory to get each resource's
+    # OpenAPI definition.
+    for dirpath, dirnames, files in os.walk("openapi_resource_definitions"):
+        # If dirnames is an empty list, it means we are at the end of a branch in the directory
+        # tree, which means the yaml files for one of the resources should be in this
+        # directory.
+        if not dirnames:
+            # Get the name of the resource that we're processing.
+            # dirpath will look something like:
+            #
+            # openapi_resource_definitions/individual_resources/Attachment
+            #
+            # So the resource name will be the last item of dirpath.
+            resource_name = os.path.split(dirpath)[-1]
+            # Iterate through the list of definition files for each HTTP method inside that
+            # resource's folder.
+            for def_file in files:
+                # Get the path to the definition file.
+                path_to_file = os.path.join(dirpath, def_file)
+                # Get the type of HTTP method from the file name by removing the ".yaml" file
+                # extension from the string.
+                http_method = def_file[:-5]
```

Comment:
Maybe, to make this clearer and self documenting, we could write:

    http_method = def_file[:-len(".yaml")]

Reply:
Oh, awesome! Didn't think of it that way. Updated